js/edit-slide-admin.js

```diff
@@ -61,7 +61,27 @@ jQuery( document ).ready( function( $ ) {
 	$( '#slides' ).on( 'click.add-data', '.button.add-data', function( e ) {
 		var table_body = $(this).closest( 'table.slide-data-attributes-table' ).find( 'tbody' );
 		var slide_index = $(this).closest( '.stuffbox' ).find( 'input[name="slide-index"]' ).val();
-		var data_row = '<tr><td class="left newdataleft"><input type="text" name="slide-data[' + slide_index + '][]"></td><td><input type="text" name="slide-data-value[' + slide_index + '][]"></td></tr>';
+		var row_id = Math.floor(Math.random() * 100000000)
+		var data_row = '<tr>';
+		data_row += '<td class="left newdataleft">';
+		data_row += '<input type="text" name="slide-data[' + slide_index + '][]" list="slide-data-' + row_id + '">';
+		data_row += '<datalist id="slide-data-' + row_id + '">';
```

Comment:
Can we just add a single datalist outside of this (or add it only if it doesn't already exist somewhere) and then reuse it from each input? My understanding is that something like this is perfectly acceptable:
```
<datalist id="mylist">
	<option value="one">
	<option value="two">
	<option value="three">
</datalist>
<input list="mylist" name="test-one">
<input list="mylist" name="test-two">
<input list="mylist" name="test-three">
```

Reply:
Fair point. I'll get some time to take a look at fixing that up tonight or tomorrow night. Good spot though!

Reply:
Took a little longer than expected. PR updated.